design/libraries/README.md

```diff
@@ -0,0 +1,10 @@
+Library Design Guidance
+=======================
+
+This directory contains some documentation of recommendations for
+library authors who want to make IPLD libraries in a new language
+(or, perhaps for readers who want to understand an existing library better).
+
+Some of the information expressed here comes down to opinions moreso than specification;
+what is good ergonomics may vary wildly per language, so take these as
+recommendations rather than strictures.
```

Comment:
I think a preface here about the perspective this is written from would be appropriate. Some of the language below applies very specifically to a narrow set of languages (the para on `Node` as an interface for example). But that's OK if we make it clear up-front where this is coming from.

Something like:

> These design guidelines are primarily based on the experience of building, and rebuilding IPLD libraries in Go and reflecting on the limitations of implementations that have existed in both Go and JavaScript and the implications of those limitations on the potential and feature-set of IPLD. The language used in these guidelines are reflective of a Go programming perspective but apply broadly to most strongly typed languages. Loosely and untyped languages will need to interpret these guidelines appropriately while extracting the key concepts.

Reply:
Added a large block of caveats to the top of the doc.

I almost wonder if we won't end up making a habit of repeating chunks of these at the top of most of our documents.  Benedictions on one page or another don't really seem to carry over to even their most proximate siblings when readers jump into one doc or other without following any path to get there that we anticipated.